apps/_pomodoro_app/realtime/application/CyclicQueue.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Cyclic queue is a queue which resets to it's original position when emptied
+ * Master queue is immutable
+ */
+class CyclicQueue {
```

Comment:
it's probably better to put it inside `utils/queue`. Like https://github.com/8iq/nodejs-hackathon-boilerplate-starter-kit/tree/master/apps/_example05app/utils `url.utils`

Reply:
I agree on a point that this should be put in `utils` — this is just a data structure, it does not describe business logic of the application, but I don't want to rename it to queue since this is not a regular queue and I don't want to let ambiguous naming.

Fixed by putting in `utils/cyclicQueue`